File: 프로그래머스/LV2/코딩테스트 연습/2017 팁스타운/짝지어 제거하기.cpp
#include <iostream>
#include<string>
#include <stack>
using namespace std;

int solution(string s)
{
    stack<int> stack;

    for(char c : s)
    {
        if(!stack.empty() && stack.top() == c)stack.pop();
        else stack.push(c);
    }
    return stack.empty() ? 1 : 0;
}

File: 프로그래머스/LV2/코딩테스트 연습/2017 팁스타운/짝지어 제거하기.py
def solution(s):
    stack = []
    top = -1
    for c in s:
        if top != -1 and stack[top] == c:
            stack.pop()
            top -= 1
        else:
            stack.append(c)
            top += 1
    return 1 if top == -1 else 0 